packages/main/src/components/AnalyticalTable/hooks/useSelectionChangeCallback.ts

```diff
@@ -0,0 +1,36 @@
+import { enrichEventWithDetails } from '@ui5/webcomponents-react-base';
+import { useEffect } from 'react';
+import { TableSelectionMode } from '../../../enums';
+
+export const useSelectionChangeCallback = (hooks) => {
+  hooks.useControlledState.push((state, { instance }) => {
+    const { selectedRowPayload, selectedRowIds } = state;
+    const { rowsById, webComponentsReactProperties, selectedFlatRows } = instance;
+    useEffect(() => {
+      if (selectedRowPayload) {
+        const { event: e, row: selRow } = selectedRowPayload;
+        const row = rowsById[selRow.id];
+        const payload = {
+          row: row,
+          isSelected: row.isSelected,
```

Comment:
this change is causing a regression issue 

https://user-images.githubusercontent.com/57886348/197931563-b904dc8f-7cbe-4cb4-a4f9-56735d0b4780.mp4